src/main/java/com/example/training/proxy/AccidentsProxyService.java

```diff
@@ -1,29 +1,33 @@
 package com.example.training.proxy;
 
+import com.example.training.client.AccidentsClient;
 import com.example.training.model.entity.Accident;
+import feign.Feign;
+import feign.jackson.JacksonDecoder;
+import feign.okhttp.OkHttpClient;
+import feign.slf4j.Slf4jLogger;
 import org.springframework.stereotype.Service;
-import org.springframework.web.client.RestTemplate;
 
 @Service
 public class AccidentsProxyService {
 
-    private final RestTemplate restTemplate;
-    private final String wiremockUrl1 = "http://localhost:8081";
+    private final AccidentsClient accidentsClient;
 
-    public AccidentsProxyService(RestTemplate restTemplate) {
-        this.restTemplate = restTemplate;
+    public AccidentsProxyService() {
+        this.accidentsClient = Feign.builder()
+                .client(new OkHttpClient())
+                .decoder(new JacksonDecoder())
+                .logger(new Slf4jLogger(AccidentsClient.class))
+                .logLevel(feign.Logger.Level.FULL)
+                .target(AccidentsClient.class, "http://localhost:8081");
```

Comment:
As per my previous comment, using RequestLine instead of defining a FeignClient bean leads to this. Even using this strategy to create a client, it would be better to have it in a configuration class exposing new beans, not like this. Also, bad practice hardcoding the base URL for the service in the code. It should be provided by a parameter. In this exercise we don't have multiple environments but bear in mind most projects need to set URLs based on the environment. Also, anyway, harcoded config is a bad practice.